### ISSUE TITLE
Rmarkdown Issue

### ISSUE THREAD

**Issue by abhayguptaastellas** [rstudio/rmarkdown#2414]
When users trying to launch R Markdown file from(File --> New File --> R Markdown option) they are getting the error(please see screenshots). Can you please check what could be the issue.
![MicrosoftTeams-image](https://user-images.githubusercontent.com/111520548/192203499-c7626ffb-0003-4f4b-92e6-86ece5bdb2eb.png)

The version we currently have is 2.7. Please see below screenshot.
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/111520548/192203709-fea7d7bb-8e32-4a7e-b2d3-e303088dd3a2.png)

Can you please let us know how we can make Rmarkdown to use version 2.7 that we have installed on our system as currently it is asking for v2.10
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/111520548/192203925-8c2ce39e-a7e9-41ea-a4fe-a44fe17c2d28.png)

**Comment by cderv** [comment]
Sometimes R packages or tools comes with minimal requirements. 

Did you upgrade RStudio IDE recently ? Recent IDE comes with a **rmarkdown** package requirement of minimal version 2.10
This means they won't work with 2.7 unfortunately. 
This is the package requirements for RStudio Workbench that it seems you are using : https://docs.rstudio.com/ide/server-pro/r_package_dependencies/r_package_dependencies.html

Also are you on a closed system ? Can't you install newer version of Package ? 
I see in the screenshot that your CRAN url is not configured, and that is why you get the error of new version required not being available. 

Configuring a correct CRAN url would allow you to install the required version. 


Anyway, this is not an issue with **rmarkdown** package. You should ask your support about your environment, or if about RStudio issues, the community is a good place https://community.rstudio.com/c/rstudio-ide/9

**Comment by vatanks007** [comment]
Hello cderv, Hello yihui,

We have not upgraded RStudio IDE. Yes we are on closed system. Right now we donot want to upgrade anything but would like to use what we have by configuring things correctly. As you said CRAN url is not configured, how did you know that? how can we configure it?
I understand you are saying there is no problem with RMarkdown package but may be with configuration - correct?

Thanks

**Comment by cderv** [comment]
> We have not upgraded RStudio IDE. Yes we are on closed system. 

Closed system meaning not external internet connection ? Will you be able to install CRAN packages ? 

> Right now we donot want to upgrade anything but would like to use what we have by configuring things correctly

With the issue you have (RStudio IDE asking for required package version), it seems you have updated something (IDE , R, packages...). If it was working before, something changed and you need to reset things correctly. 

> As you said CRAN url is not configured, how did you know that? how can we configure it?

It is in the warning of the first screenshot. You can configure it by using R options or directly in the RStudio IDE options (Look at your own IDE configuration option in Menu or at admin conf at https://docs.rstudio.com/ide/server-pro/r_sessions/package_installation.html#cran-repositories) 

> I understand you are saying there is no problem with RMarkdown package but may be with configuration - correct?

My understanding of the issue is: 

* You are using a RStudio IDE version that requires some minimum R packages version
* Currently your R installation can't find the packages in the required versions

2 things I have in mind: 
* Packages are installed but something is wrong in the way IDE does not find them where they should. Wrong Library Path, or something like that. 
* Packages are not installed (like with a new R versions) and you need to reinstall them - for that you need to have access to a CRAN repo from your environment, either one on the internet or one on your local network set by your IT. 

Anyway, no issue with the packe itself IMO. So you need to ask you admins to help with correctly setting your environment.

**Comment by vatanks007** [comment]
Hello Dervieux,

Thank you for your response.

I suspect this also, that current R installation not able to find required packages in this version.
Is it possible for you to provide the steps where to navigate for setting up the library path so that R would be able to pickup correct packages for RMarkdown?


--
Thanks and Regards,
Vatan Singh


________________________________

**Comment by cderv** [comment]
> Is it possible for you to provide the steps where to navigate for setting up the library path so that R would be able to pickup correct packages for RMarkdown ?

It depends how you are setting this up usually, using R options globally or per user, or using RStudio Workbench configurations. 

Look in the admin doc for workbench: https://docs.rstudio.com/ide/server-pro/r_sessions/package_installation.html
and also. 

Also it depends if you are in this state because of a R update. Some info in https://rstats.wtf/maintaining-r.html#how-to-transfer-your-library-when-updating-r

Basically look at `.libPaths()` to see which are the libray paths where packages are searched for. Then configure using ENV VAR or reinstall in the correct place

**Comment by vatanks007** [comment]
Thanks. I will look into it and come back in case any thing else required.


--
Thanks and Regards,
Vatan Singh


________________________________
